### ISSUE TITLE
PIDE crashes with a combination of Program and Print.

### ISSUE THREAD

**Issue by jesper-bengtson** [hannesm/Kopitiam#93]
The following program crashes

```
Require Import Coq.Program.Tactics.

Class Test := {
  a : False
}.

Program Instance test : Test := {
  a := _
}.

Print nat.
```

Removing either Program or Print will make this work.

**Comment by ale-f** [comment]
The program above seems to break STM somehow. Coqoon's debug log shows the following error just before the `coqtop` process stops:

```
debug/pide: ? stderr [[master:5] Fatal marshal error in query
]]
```

It seems to be a problem limited to STM, though, as plain `coqtop` accepts this program with no complaints.

**Comment by gares** [comment]
It is surely a bug in STM when used in combination with PIDE.

**Comment by gares** [comment]
This is closed now!

**Comment by ale-f** [comment]
Indeed it is; fixed in commit coq/coq@1834d60f373fb63c32d179f2cf03ee4da7880bb0.
